guides/v2.1/ext-best-practices/tutorials/dynamic-row-system-config.md

```diff
@@ -0,0 +1,75 @@
+---
+group: extension-best-practices
+subgroup: 02_Extension-Coding
+title: Creating a dynamic row system config
+menu_title: Creating a dynamic row system config
+menu_order: 1010
+functional_areas:
+  - Standards
+---
+
+This tutorial shows you how to add a new dynamic rows system configuration in the {% glossarytooltip 18b930cf-09cc-47c9-a5e5-905f86c43f81 %}Magento admin{% endglossarytooltip %}, by extending the [Magento/Config/Block/System/Config/Form/Field/FieldArray/AbstractFieldArray][0]{:target="_blank"} class. 
+
+## Step 1: Add a new system field 
+
+> `etc/adminhtml/system.xml`
+
+```xml
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Config:etc/system_file.xsd">
+    <system>
+        <section id="general" translate="label" type="text">
+            <group id="quantity_ranges" translate="label" type="text" sortOrder="10" showInDefault="1" showInWebsite="1" showInStore="1">
+                <label>Quantity Ranges</label>
+                <field id="ranges" translate="label" sortOrder="5" showInDefault="1" showInWebsite="1" showInStore="1">
+                    <label>Ranges</label>
+                    <frontend_model>[vendor]\[module]\Block\Adminhtml\Form\Field\Ranges</frontend_model>
+                    <backend_model>Magento\Config\Model\Config\Backend\Serialized\ArraySerialized</backend_model>
+                </field>
+            </group>
+        </section>
+    </system>
+</config>
+```
+
+This code adds a new system config in the following location:
+
+``Stores/Configuration/General [General]/Quantity Ranges``
```

Comment:
This path doesn't read right, and I think there are too many ` (remove one from each side).

Reply:
The ` have been removed.

Reply:
<img width="679" alt="Screen Shot 2019-05-21 at 8 05 05 AM" src="https://user-images.githubusercontent.com/14031342/58098492-4e964c00-7b9f-11e9-8140-e29a571e0e93.png">

Reply:
I don't really get why the current rendering is not good, like
`Stores/Configuration/General [General]/Quantity Ranges` ?
Maybe we should use a different separator, like `->`. What do you think?

Reply:
Why is `General` in brackets? It doesn't make sense. It should be `Stores/Configuration/General/QuantityRanges` (also with no spaces, as it's unlikely the actual path has spaces---am I wrong?)

Is this the actual directory of a file, or a clickable path to get somewhere in the UI?

Directory: https://wiki.corp.magento.com/display/MDOC/OMS+Retros
Clickable path: `**Stores** > **Configuration** > **General** > **QuantityRanges**

Reply:
Regarding the `General` in brackets, I wanted to specify the tab, but agree that it is not the best option. After checking some other examples, looks like a better solution is to show it as:
`**Stores** > **Configuration** > **General** > **Quantity Ranges**`
I was checking this one as example:
https://devdocs.magento.com/guides/v2.3/frontend-dev-guide/templates/template-email.html#customize-content

Reply:
So, I don't have the UI open, but things that you actually click on should be bolded, and sections/areas that options lie within (but that don't require clicking) should not be bolded. The words should also be capitalized and spelled as exactly as in the UI.

Can you confirm that this applies to your changes to the clickable path?

Reply:
Yes, it is exactly as it is described. The clickable menu items are bolded and the sections (tabs) are not. Also the menu items are spelled and capitalized as they actually are.
Here the green highlighted are clickable items, and the blue ones are not.
![Xnip2019-05-21_21-58-21](https://user-images.githubusercontent.com/15868188/58123037-9951ab80-7c13-11e9-8418-1b4fdfe7d8a0.jpg)

Reply:
Great! Thank you for this 💯